langkit/templates/pkg_implementation_body_ada.mako

```diff
@@ -3704,9 +3704,12 @@ package body ${ada_lib_name}.Implementation is
       if Ent.Node /= null then
          declare
             Node_Image : constant Text_Type := Short_Text_Image (Ent.Node);
+            File_Name : constant Text_Type := To_Text (
+              Ada.Directories.Simple_Name (Get_Filename (Unit (Ent.Node))));
          begin
             return
-            (if Ent.Info.Rebindings /= null
+            (if Ent.Info.Rebindings /= null and File_Name /= "__prelude"
```

Comment:
Should be `and then` (we almost never use `and` in Ada, except for bitwise operations)